### PR TITLE
Document the source documentation + document some modules

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

**Comment by sylvestre** [comment]
To see the result:
https://sylvestre.github.io/coreutils/coreutils/

Will be generated on the first merge

**Comment by rivy** [comment]
@sylvestre , I see that you're working on docs automation and that you initiated a 'gh-pages' branch.

Unfortunately, my experience is that the separate and parallel 'gh-pages' branch is not the best way to go... it confuses the history display of the project making branch usage more complicated. Alternatively, we should be able to integrate the docs into the main project in a specific folder and point GitHub at that folder (see <https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch>).

Do you mind coding toward that instead?

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        profile: minimal
+
+    - name: Build Documentation
+      run: cargo doc --no-deps --all-features --workspace
+
+    - name: Deploy Docs
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./target/doc
```
Comment (rivy) [inline .github/workflows/docs.yml:29]: Deploy to 'master' or even another repository completely (eg, 'coreutils.gh-pages')?

**Review by rivy — CHANGES_REQUESTED** [review]
Good start!

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Docs
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
```
Comment (rivy) [inline .github/workflows/docs.yml:9]:
```suggestion
on:
  # * only trigger on pull request closed to specific branches
  # ref: https://github.community/t/trigger-workflow-only-on-pull-request-merge/17359/9
  pull_request:
    branches: [ master ]
    types: [ closed ]

jobs:
  docs:
    if: github.event.pull_request.merged == true ## only for PR merges
```

Trigger only on PR merges (GitHub has some build limits for the site)?

File: .github/workflows/docs.yml (hunk 1 above)
Comment (sylvestre) [reply .github/workflows/docs.yml:29]: it uses gh-pages by default as a new branch

File: .github/workflows/docs.yml (hunk 2 above)
Comment (sylvestre) [reply .github/workflows/docs.yml:9]: sure

**Comment by sylvestre** [comment]
@rivy so, you prefer a new repo or a specific branch is good enough?

**Comment by sylvestre** [comment]
@rivy works for you?

**Comment by rivy** [comment]
Sure; it's fine.
I'll look into automation for building/moving the doc output to an alternate repository so that it doesn't clutter and bloat this one, if that's ok with you?

**Comment by rivy** [comment]
@sylvestre , I'm looking at this more closely, and I'm not sure why we need it. It looks like all this documentation is produced automatically on *docs.rs* whenever we publish. Why would we need to generate and hold the product here?

**Comment by sylvestre** [comment]
> I'll look into automation for building/moving the doc output to an alternate repository so that it doesn't clutter and bloat this one, if that's ok with you?

Sure, thanks!

**Comment by sylvestre** [comment]
I went ahead with your solution.
Generated here:
https://github.com/uutils/coreutils-docs
and published here:
https://uutils.github.io/coreutils-docs/coreutils/

So, this PR is to document the change!

**Comment by sylvestre** [comment]
uucore's doc is pretty cool: https://uutils.github.io/coreutils-docs/uucore/index.html :)

**Comment by sylvestre** [comment]
It run last night: https://github.com/uutils/coreutils-docs/actions/runs/1164884695

**Review by miDeb — APPROVED** [review]
This is great! Now the only thing that's left is writing more documentation ;)

**Pull request merged**